src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+    const separatorToString = String(separator);
+
+    for (const item of this) {
+      if (item === null || item === undefined) {
+        joinedString += '' + separatorToString;
+      } else {
+        joinedString += item + separatorToString;
+      }
+    }
+
+    // eslint-disable-next-line max-len
+    return joinedString.slice(0, joinedString.length - separatorToString.length);
```

Comment:
a little bit complicated as I like)))) but mentors dont:(

Reply:
The separator's length can be more than 1, so in this case I decided that I need exactly separator.length instead of just 1